scanner/scanner.go

```diff
@@ -0,0 +1,150 @@
+package scanner
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/SkynetLabs/malware-scanner/clamav"
+	"github.com/SkynetLabs/malware-scanner/database"
+	"github.com/SkynetLabs/skynet-accounts/build"
+	"github.com/sirupsen/logrus"
+	"gitlab.com/NebulousLabs/errors"
+)
+
+var (
+	// sleepBetweenScans defines how long the scanner should sleep after
+	// scanning the DB and not finding any skylinks to scan.
+	sleepBetweenScans = build.Select(
+		build.Var{
+			Dev:      30 * time.Second,
+			Testing:  100 * time.Millisecond,
+			Standard: 10 * time.Second,
+		},
+	).(time.Duration)
+)
+
+// Scanner provides a convenient interface for working with ClamAV
+type Scanner struct {
+	staticCtx    context.Context
+	staticDB     *database.DB
+	staticClam   *clamav.ClamAV
+	staticLogger *logrus.Logger
+}
+
+// New returns a new Scanner with the given parameters.
+func New(ctx context.Context, db *database.DB, clam *clamav.ClamAV, logger *logrus.Logger) *Scanner {
+	return &Scanner{
+		staticCtx:    ctx,
+		staticDB:     db,
+		staticClam:   clam,
+		staticLogger: logger,
+	}
+}
+
+// SweepAndScan sweeps the DB for new skylinks, locks them, scans them,
+// and updates their records in the DB. The method returns a boolean flag that
+// instructs the caller to sleep after this call when there are no records to
+// be scanned.
+func (s Scanner) SweepAndScan(abort chan bool) (sleepAfter bool) {
```

Comment:
typically our engineering patterns leave the decision-making to the caller. So IMO this method should return the error and the caller can decide if they should sleep or not.  

I also don't see a huge logic benefit for handling that sleep decision in this method since there is just the one condition which returns true.

Reply:
That makes sense. Changed to return error, so the caller can decide.